lib/buffer.js

```diff
@@ -146,9 +146,9 @@ function Buffer(arg, encodingOrOffset, length) {
   // Common case.
   if (typeof arg === 'number') {
     if (typeof encodingOrOffset === 'string') {
-      throw new Error(
-        'If encoding is specified then the first argument must be a string'
-      );
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', {
+        msg: 'The first argument must be of type string.'
```

Comment:
I believe these should be passed parameters which provide info about the arg and expected type as opposed to a fixed string.  See  invalidArgType(name, expected, actual) in internal/errors.js